tests/rawkv.go

```diff
@@ -0,0 +1,424 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+
+	"github.com/tikv/client-validator/mocktikv"
+	"github.com/tikv/client-validator/stub"
+	"github.com/tikv/client-validator/validator"
+)
+
+type testRawKV struct{}
+
+func (t testRawKV) newCluster(ctx validator.ExecContext) *mocktikv.Cluster {
+	cluster, err := mocktikv.NewCluster(*mockTiKVAddr)
+	ctx.AssertNil(err)
+	return cluster
+}
+
+var _ = validator.Feature("rawkv.new", "create a rawkv client", nil, testRawKV{}.checkClientCreate)
+
+func (t testRawKV) checkClientCreate(ctx validator.ExecContext) validator.FeatureStatus {
+	cluster := t.newCluster(ctx)
+	defer cluster.Close()
+	client, err := stub.NewRawClientStub(*clientProxyAddr, cluster.PDAddrs())
+	if err != nil {
+		return errToFeatureStatus(err)
+	}
+	defer client.Close()
```

Comment:
Do we need to put this line before if statement?

Reply:
I think not, if err is no nil, client should be nil instead, so we cannot close it.